core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -207,30 +218,51 @@ private FunctionResolver current_date() {
   }
 
   /**
-   * Specify a start date and add a temporal amount to the date.
+   * A common signature for `date_add` and `date_sub`.
+   * Specify a start date and add/subtract a temporal amount to/from the date.
    * The return type depends on the date type and the interval unit. Detailed supported signatures:
-   * (STRING/DATE/DATETIME/TIMESTAMP, INTERVAL) -> DATETIME
-   * (DATE, LONG) -> DATE
-   * (STRING/DATETIME/TIMESTAMP, LONG) -> DATETIME
+   * (DATE/DATETIME/TIMESTAMP/TIME, INTERVAL) -> DATETIME
+   * MySQL has these signatures too
+   * (DATE, INTERVAL) -> DATE                                     // when interval has no time part
+   * (TIME, INTERVAL) -> TIME                                     // when interval has no date part
+   * (STRING, INTERVAL) -> STRING         // when argument has date or datetime string,
+   *                                      // result has date or datetime depending on interval type
    */
-  private DefaultFunctionResolver add_date(FunctionName functionName) {
-    return define(functionName,
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval),
-            DATETIME, STRING, INTERVAL),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval), DATETIME, DATE, INTERVAL),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval),
+  private Stream<SerializableFunction<?, ?>> get_date_add_date_sub_signatures(
+      SerializableTriFunction<FunctionProperties, ExprValue, ExprValue, ExprValue> function) {
+    return Stream.of(
```

Comment:
Does Datetime required? I found you post same concern in https://github.com/opensearch-project/sql/issues/1176.

Reply:
Unfortunately, yes, unless we have 4 types for date/time.